subsys/nrf_security/src/drivers/cracen/cracenpsa/src/platform_keys/platform_keys.c

```diff
@@ -673,7 +681,7 @@ psa_status_t cracen_platform_keys_provision(const psa_key_attributes_t *attribut
 
 	nrf_mramc_config_set(mramc, &mramc_config_write_enabled);
 
-	memcpy(key.sicr.nonce_addr, &key.sicr.nonce, sizeof(key.sicr.nonce));
+	memcpy(key.sicr.nonce_addr, &key.sicr.nonce, sizeof(key.sicr.nonce[0]));
```

Comment:
Do we sizeof the `[0]` member on purpose?

Reply:
it probably is. The nonce buffer contains 3 uint32_t elements, but that's only because the function the set_nonce function expects that. Only 4 bytes of data are being used, the rest is padded with zeros. So when the data are stored in sicr only the first four bytes are being stored.

Reply:
Could we update the struct to that the nonce value and padding are distinct members?

Reply:
Sure, I can update it

Reply:
Actually, splitting nonce and padding complicates the usage of it.
I'll add comments in the code where we use it instead, explaining why we use sizeof of [0]